src/software/ai/evaluation/calc_best_shot_test.cpp

```diff
@@ -177,7 +177,7 @@ TEST(CalcBestShotTest, calc_best_shot_on_enemy_goal_with_all_shots_blocked_by_ob
                                                       world.enemyTeam(), shooting_robot);
 
     // We should not be able to find a shot
-    ASSERT_FALSE(result);
+    EXPECT_NEAR(result->getOpenAngle().toRadians(), 0, 0.001);
```

Comment:
Good point, changed to EXPECT_EQ

Reply:
Another thought, why don't we return an `std::optional` so we can return `std::nullopt` if no shot is found? Seems to make more sense then just checking if the open angle is zero degrees exactly. I'm OK with it as-is, because zero degrees is still pretty clear, but just a suggestion for your consideration.

Reply:
angleSweep originally returned a nullopt if the shot is 0 degrees. 

I changed it because I didn't think it was the job of the function to rate the shot.

IMO, whoever is using the shot eval functions should be checking that the shot is 'good enough' for the scenario.

Thoughts anyone else?

Reply:
tl;dr I'm leaning towards leaving as-is and expecting the angle to be checked, but there are pros and cons

Not returning an optional is a bit easier to work with, because it reduces the number of checks that need to take place, which arguably can be redundant. It's not so much the angle being 0 that's the issue, it's "What is the placeholder value of the shot direction / target if no shot is found". People could still not check the angle correctly and end up shooting at an unexpected place.

However I've seen the same mistake made back when we did use an optional. Because people knew (maybe) that if the optional returned a value, the angle was non-zero, they would only check the optional and not the angle before taking a shot, which also led to undesired behavior. Using an optional means you don't have "placeholder values" or "undefined behavior" if no shot is found, where it doesn't necessarily make sense for the shot to have a direction because there "is no shot".

Overall I think the same mistakes can be made with both, and I'm fine with things as they are (not returning an optional).

Reply:
Ok, also fine with things as they are now. Marking as resolved.